datarobot_batch_scoring/network.py

```diff
@@ -33,6 +35,12 @@ def _request(self, request):
                 callback = request.hooks['response'][0]
             response = FakeResponse(400, 'No Response')
             callback(response)
+        finally:
+            request.data = None
+            prepared.data = None
+            del request
+            del prepared
+            gc.collect()
```

Comment:
this is quite costly after every request -- please note that the process running the thread pool is performance critical

furthermore, i can still see memory leakage... i suppose we either have circular references somewhere in our requests callback.